add example of response encryption (#536)
openid/OpenID4VP: openid-4-verifiable-presentations-1_0.md

Using what it supports and its preferences, the Wallet selects the public key to encrypt the Authorization Response based on information about each key, such as the `kty` (Key Type), `use` (Public Key Use), `alg` (Algorithm), and other JWK parameters.
The JWE `alg` algorithm used MUST the `alg` value of the chosen `jwk`, if present, or otherwise make sense to use with the selected key.
If the selected public key contains a `kid` parameter, the JWE MUST include the same value in the `kid` JWE Header Parameter (as defined in [@!RFC7516, Section 4.1.6]) of the encrypted response. This enables the Verifier to easily identify the specific public key that used to encrypt the response.

> The JWE `enc` content encryption algorithm used is obtained from the `authorization_encrypted_response_enc` parameter of client metadata, such as the `client_metadata` request parameter, allowing for the Wallet to use a content encryption algorithm of its choosing when not explicitly set.

Reviewer: I'd say the practical effect of this is probably that verifiers have to set this value as they can't risk the wallet returning one they don't support?

Author: Practically there are only 6 defined content encryption algorithms defined https://www.iana.org/assignments/jose/jose.xhtml and I am not aware of libraries/platforms that don't support most or all of them (thought of course my rough observation is not an objective measure). 

The current JOSE HPKE work is making some IMO very questionable choices with treatment of algorithms. The current draft has what amounts to a null content encryption algorithm that only can/should be used in conjunction with a particular HPKE based key management alg. 

With this wording, inspired by on informal questions you'd asked about the PR that proceeded this, I was trying to make room to better allow agility with respect to wherever the shenanigans with JOSE HPKE might land. That was probably a mistake. And definitely shouldn't have tied to adding a (hopefully) helpful example. 

This would undo that but fix a spelling error:

```suggestion
The JWE `enc` content encryption algorithm used is obtained from the `authorization_encrypted_response_enc` parameter of client metadata, such as the `client_metadata` request parameter, allowing for the for the default value of `A128CBC-HS256` when not explicitly set.
```

Reviewer 2: we agreed on A128GCM during post-IIW mtg

Author: > we agreed on A128GCM during post-IIW mtg

That's what I thought but then the merged #523 had `A128CBC-HS256` in this part of the text. I *think* maybe the #523 had it in two places but then only one was changed during the post-IIW mtg. Anyway, how's about:

```suggestion
The JWE `enc` content encryption algorithm used is obtained from the `authorization_encrypted_response_enc` parameter of client metadata, such as the `client_metadata` request parameter, allowing for the for the default value of `A128GCM` when not explicitly set.
```

Reviewer 2: oh, then we probably missed it. thanks for catching! lgtm

Reviewer 3: Yes, this was discussed on Friday, but to the extent that we're using the same parameter name as JARM, it should have the same meeting.  [JARM says](https://openid.net/specs/oauth-v2-jarm-final.html#name-client-metadata) "the default for this value is A128CBC-HS256."  We should do the same.

Related to this, a recorded action for me in the minutes from Friday was:
> Mike will share a link to text that says gcm default support is required for JARM.

In fact, A128CBC-HS256 is what's required, both for JARM and in the [IANA JSON Web Signature and Encryption Algorithms registry](https://www.iana.org/assignments/jose/jose.xhtml#web-signature-encryption-algorithms).  I'll correct that on-list.  Whereas support for A128GCM is recommended in the registry, not required.

    {
     "kty":"OKP","kid":"jc","use":"enc","crv":"X25519", "alg":"ECDH-ES",
     "x":"WPX7wnwq10hFNK9aDSyG1QlLswE_CJY14LdhcFUIVVc"
    },
    {
     "kty":"EC","kid":"lc","use":"enc","crv":"P-384","alg":"ECDH-ES",
     "x":"iHytgLNtXjEyYMAIGwfgjINZRmLfObYbmjPhkaPD8OiTkJtRHjegTNdH31Mxg4nV",
     "y":"MizXWSqNB7sSt_SNjg3spvaJnmjB-LpxsPpLUaea33rvINL3Mq-gEaANErRQpbLx"
    },
    {
     "kty":"OKP","kid":"bc","use":"enc","crv":"X448","alg":"ECDH-ES",
     "x":"pK5IRpLlX-8XcsRYWHejpzkfsHoDOmAYuBzAC7aTpewWOw_QFHSa64t9p2kuommI8JQQLohS2AIA"
    }
   ]
  },
  "authorization_encrypted_response_enc": "A128GCM"
 }
}
```

A non-normative example response to the above request, having been encrypted to the first key, might look like the following
(with added line breaks for display purposes only):

```
{
 "response" : "eyJhbGciOiJFQ0RILUVTIiwiZW5jIjoiQTEyOEdDTSIsImtpZCI6ImFjIiwiZXBrIjp7Imt
    0eSI6IkVDIiwieCI6Im5ubVZwbTNWM2piaGNhZlFhUkJrU1ZOSGx3Wkh3dC05ck9wSnVmeVlJdWsiLCJ5I
    joicjRmakRxd0p5czlxVU9QLV9iM21SNVNaRy0tQ3dPMm1pYzVWU05UWU45ZyIsImNydiI6IlAtMjU2In1
    9..uAYcHRUSSn2X0WPX.yVzlGSYG4qbg0bq18JcUiDRw56yVnbKR8E7S7YlEtzT00RqE3Pw5oTpUG3hdLN
    4taHZ9gC1kwak8JOnJgQ.1wR024_3-qtAlx1oFIUpQQ"

> }

Author: I meant to, but apparently forgot, to include the private key too so folks playing along at home could decrypt this.

Author: now added
